### PR TITLE
Use Google Play Services to get the advertising id

### PR DESCRIPTION
**Pull request by hbbtstar**
This should address https://github.com/Parsely/parsely-android/issues/5.

Confirmed working in the ParselyExample app (which has been updated due to that testing).

I should point out there is a possibility, if a track call fires extremely quickly, that the adKey is still null for the first pageview: but it's a consequence of getting the IdInfo in an asynctask, which the Android docs recommend. We could block on getting the ad info, but that seems like a bad solution to me.

I think the rare possibility of one pageview having a different uuid is worth it, considering how hard it was to replicate in code (the adkey get took just a couple hundred milliseconds), but if we really want to ensure that doesn't happen the only other way I can think of other than blocking on that call (which we shouldn't do) it is to loop through the queue and replace the deviceInfo if adKey is set before we flush it.

### CODE REVIEW

File: parsely/parselyandroid/ParselyTracker.java [hunk 1]
```diff
@@ -331,10 +337,12 @@ protected ParselyTracker(String apikey, int flushInterval, String urlref, Contex
 
         this.apikey = apikey;
         this.uuidkey = "parsely-uuid";
+        this.adKey = null;
+        // get the adkey straight away on instantiation
+        new GetAdKey(c).execute();
         this.flushInterval = flushInterval;
         this.storageKey = "parsely-events.ser";
         //this.rootUrl = "http://10.0.2.2:5001/";  // emulator localhost
-        this.rootUrl = "http://srv.pixel.parsely.com/";
```
Comment (emmett9001) [inline parsely/parselyandroid/ParselyTracker.java]: Why is this removed? Looks like it's still important.

File: parsely/parselyandroid/ParselyTracker.java [hunk 2]
```diff
@@ -469,4 +477,38 @@ protected Void doInBackground(Void... params) {
...
+            mContext = context;
+        }
+
+        @Override
+        protected String doInBackground(Void... params) {
+            AdvertisingIdClient.Info idInfo = null;
+            String advertId = null;
+            try {
+                idInfo = AdvertisingIdClient.getAdvertisingIdInfo(mContext);
+            } catch (GooglePlayServicesRepairableException |IOException|GooglePlayServicesNotAvailableException e) {
```
Comment (emmett9001) [inline parsely/parselyandroid/ParselyTracker.java]: Strange spacing around operators here

File: parsely/parselyandroid/ParselyTracker.java [hunk 3]
```diff
@@ -469,4 +477,38 @@ protected Void doInBackground(Void... params) {
...
+                PLog("No Google play services or error! falling back to device uuid");
+                // fall back to device uuid on google play errors
+                advertId = getSiteUuid();
```
Comment (emmett9001) [inline parsely/parselyandroid/ParselyTracker.java]: This works without `this` because this class is nested within `ParselyTracker`, right?

File: parsely/parselyandroid/ParselyTracker.java (hunk 1 above)
Comment (hbbtstar) [reply parsely/parselyandroid/ParselyTracker.java]: it definitely is, not sure how I axed it and then missed it! adding back.

File: parsely/parselyandroid/ParselyTracker.java (hunk 2 above)
Comment (hbbtstar) [reply parsely/parselyandroid/ParselyTracker.java]: good catch, will make it neater

File: parsely/parselyandroid/ParselyTracker.java (hunk 3 above)
Comment (hbbtstar) [reply parsely/parselyandroid/ParselyTracker.java]: yup, inner classes can access outer class variables like this (unless I had declared a variable in the inner class with the same name)

**Comment by hbbtstar** [comment]
in testing some more I was unable to get the adKey to not set even firing a PV as soon as the app loads (I suspect Google Play Services caches this) so I think we're good to go without doing all the complicated checking in the batch header. Updated with other comments!
